### PR TITLE
MAPFAIL Produces a Semantic Error in HANDLE CONDITION

### LINKED ISSUE #668: MAPFAIL Produces a Semantic Error in HANDLE CONDITION
## Description ##

The `MAPFAIL` exception condition produces a semantic error in `EXEC CICS`

## Steps to reproduce ##
1. Open a COBOL file
2. Add `EXEC CICS IGNORE CONDITION MAPFAIL END-EXEC.`
3. Add `EXEC CICS HANDLE CONDITION MAPFAIL(PARNAME) END-EXEC.` where PARNAME is a paragraph name
4. See a semantic error on MAPFAIL in both cases

## Expected behavior ##

No errors are shown.

## Environment ##
 - OS: All
 - IDE: All

## Additional information ##

https://www.ibm.com/support/knowledgecenter/SSGMCP_5.6.0/applications/designing/dfhp31o.html

### PR DESCRIPTION
**Pull request by grianbrcom**
ANTLR parser was changed accordingly.

Closes #668

Signed-off-by: Anton Grigorev <anton.grigorev@broadcom.com>

### CODE REVIEW

File: server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4 [hunk 1]
```diff
@@ -422,7 +422,7 @@ pf_option: PF1 | PF2 | PF3 | PF4 | PF5 | PF6 | PF7 | PF8 | PF9 | PF10 | PF11 | P
            PF18 | PF19 | PF20 | PF21 | PF22 | PF23 | PF24;
 
 /** IGNORE CONDITION */
-cics_ignore: IGNORE (CONDITION | mama | cics_resp)+;
+cics_ignore: IGNORE CONDITION? (mama | cics_resp)+;
```
Comment (temanbrcom) [inline server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4:425]: AFAIR. `CONDITION` is obligatory here. The previous implementation was incorrect.

**Comment by grianbrcom** [comment]
I'll add more conditions from https://www.ibm.com/support/knowledgecenter/SSGMCP_5.4.0/reference/commands-api/dfhp4_receivemap.html

File: server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4 (hunk 1 above)
Comment (grianbrcom) [reply server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4:425]: I'll fix it.

File: server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4 [hunk 2]
```diff
@@ -415,14 +417,14 @@ cics_handle_abend: ABEND (CANCEL | PROGRAM cics_name | LABEL cics_label | RESET
 cics_handle_aid: AID (ANYKEY (cics_label)? | CLEAR (cics_label)? | CLRPARTN (cics_label)? | ENTER (cics_label)? |
                  LIGHTPEN (cics_label)? | OPERID  (cics_label)? | pa_option (cics_label)? | pf_option (cics_label)? |
                  TRIGGER  (cics_label)? | cics_resp)*;
-cics_handle_condition: CONDITION (mama cics_label? | cics_resp)+;
+cics_handle_condition: CONDITION ((cics_conditions | mama) cics_label? | cics_resp)+;
```
Comment (temanbrcom) [inline server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4:420]: May be simplified to `(cics_conditions | mama | cics_label | cics_resp)+`

**Review by temanbrcom — APPROVED** [review]
📟

File: server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4 (hunk 2 above)
Comment (grianbrcom) [reply server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4:420]: I'd like not to do that.
```
cics_label: LPARENCHAR paragraphNameUsage RPARENCHAR;
```
So the following line will be valid `EXEC CICS HANDLE CONDITION (10-EXIT) END-EXEC.`

Comment (temanbrcom) [reply server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4:420]: But this one is valid, isn't it?

Comment (temanbrcom) [reply server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4:420]: Sorry, you are right

**Pull request merged**